### PR TITLE
Feature/fetch handler

### PR DESCRIPTION
**Pull request by vmidyllic**
To handle  credential offer message we need to parse it and fetch credentials with a jwz auth

### CODE REVIEW

File: src/iden3comm/handlers/auth.ts [hunk 1]
```diff
@@ -12,13 +12,13 @@ import {
   ZeroKnowledgeProofRequest,
   ZeroKnowledgeProofResponse,
   ZKPPackerParams
-} from '../iden3comm';
+} from '..';
```
Comment (Kolezhniuk) [inline src/iden3comm/handlers/auth.ts:15]: add path explicitly

File: src/iden3comm/handlers/fetch.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        type: PROTOCOL_MESSAGE_TYPE.AUTHORIZATION_RESPONSE_MESSAGE_TYPE,
+        thid: offerMessage.thid ?? guid,
+        body: {
+          id: credentialInfo.id
+        },
+        from: did.toString(),
+        to: did.toString()
+      };
+
+      const msgBytes = new TextEncoder().encode(JSON.stringify(fetchRequest));
```
Comment (Kolezhniuk) [inline src/iden3comm/handlers/fetch.ts:86]: import textencoder from utils

File: src/iden3comm/handlers/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      const token = byteDecoder.decode(
+        await this._packerMgr.pack(MediaType.ZKPMessage, msgBytes, {
+          senderDID: did,
+          profileNonce: 0,
+          provingMethodAlg: proving.provingMethodGroth16AuthV2Instance.methodAlg
+        } as ZKPPackerParams)
+      );
+
+      const resp = await axios.post<CredentialIssuanceMessage>(offerMessage.body.url, token);
+      if (resp.status != 200) {
```
Comment (Kolezhniuk) [inline src/iden3comm/handlers/fetch.ts:96]: !==

**Review by Kolezhniuk — APPROVED** [review]

File: src/iden3comm/handlers/fetch.ts (hunk 2 above)
Comment (vmidyllic) [reply src/iden3comm/handlers/fetch.ts:86]: fixed

File: src/iden3comm/handlers/fetch.ts (hunk 3 above)
Comment (vmidyllic) [reply src/iden3comm/handlers/fetch.ts:96]: fixed

**Pull request merged**